android/KMEA/app/src/main/java/com/tavultesoft/kmea/logic/ResourcesUpdateTool.java

```diff
@@ -0,0 +1,492 @@
+  {
+    Integer notificationid;
+    String updateid;
+    Bundle bundle;
+  }
+
+  private Context currentContext;
+
+  private boolean updateCheckFailed = false;
+  private boolean updateFailed = false;
+  private Calendar lastUpdateCheck = null;
+  private boolean checkingUpdates = false;
+
+  private LinkedHashMap<String, OngoingUpdate> openUpdates = new LinkedHashMap<>();
+  private int failedUpdateCount = 0;
+
+  private PropertyChangeSupport propertyChangeSupport = new PropertyChangeSupport(this);
+
+
+  private AtomicInteger notificationid = new AtomicInteger(1);
+
+  private void checkResourceUpdatesInternal(boolean anInitialize) {
+
+    Runnable onSuccess = new Runnable() {
+      public void run() {
+        if(! openUpdates.isEmpty()) {
+          return;
+        }
+
+        Toast.makeText(currentContext, currentContext.getString(R.string.update_check_current), Toast.LENGTH_SHORT).show();
+        lastUpdateCheck = Calendar.getInstance();
+        SharedPreferences prefs = currentContext.getSharedPreferences(currentContext.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = prefs.edit();
+        editor.putLong(PREF_KEY_LAST_UPDATE_CHECK, lastUpdateCheck.getTime().getTime());
+        editor.commit();
+        checkingUpdates = false;
+      }
+    };
+
+    Runnable onFailure = new Runnable() {
+      public void run() {
+        if(! openUpdates.isEmpty()) {
+          return;
+        }
+
+        Toast.makeText(currentContext, currentContext.getString(R.string.update_check_unavailable), Toast.LENGTH_SHORT).show();
+        lastUpdateCheck = Calendar.getInstance();
+        updateCheckFailed = true;
+        checkingUpdates = false;
+      }
+    };
+
+    checkingUpdates = true;
+    if(anInitialize)
+      CloudRepository.shared.initializeDataSet(currentContext, this, onSuccess, onFailure);
+    else
+      CloudRepository.shared.updateDatasetIfNeeded(currentContext, this, onSuccess, onFailure);
+
+  }
+
+  public static void createNotificationChannel(Context aContext) {
+    // Create the NotificationChannel, but only on API 26+ because
+    // the NotificationChannel class is new and not in the support library
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && SEND_UPDATE_NOTIFICATIONS) {
+      CharSequence name = aContext.getString(R.string.keyboard_updates_channel);
+      String description = aContext.getString(R.string.keyboard_updates_available);
+      int importance = NotificationManager.IMPORTANCE_DEFAULT;
+      NotificationChannel channel = new NotificationChannel(ResourcesUpdateTool.class.getName(), name, importance);
+      channel.setDescription(description);
+      // Register the channel with the system; you can't change the importance
+      // or other notification behaviors after this
+      NotificationManager notificationManager = aContext.getSystemService(NotificationManager.class);
+      notificationManager.createNotificationChannel(channel);
```

Comment:
Is this the system notification? I wonder if we can also add the Keyman logo to the notification.